spec/fixtures/hiera/hiera_badkey.yaml

```diff
@@ -9,3 +9,5 @@ typical:typo::warning1:   true
 noCamelCase::warning3:    true
 no-hyphens::warning4:     true
 :picky::warning5:         true
+this_is_not_ok:
+  "%{lookup('this_is_ok'):3306}": []
```

Comment:
@ekohl @alexjfisher this fails because `:3306` is within `%{}`. This is valid YAML but not Valida hiera data. Do you have an idea how we could validate this / if puppet-syntax is the right place for this?

Reply:
Not sure...

Would also be interesting to test `alias` which wouldn't allow the port number outside the `{}` either.

Reply:
I did a bit of testing and one major problem is that the gem has a focus on validating hiera *keys*, not hiera *data*.